src/main/java/io/confluent/connect/hdfs/orc/OrcUtil.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.connect.hdfs.orc;
+
+import org.apache.hadoop.hive.ql.io.orc.OrcStruct;
+import org.apache.hadoop.hive.serde2.io.ByteWritable;
+import org.apache.hadoop.hive.serde2.io.DateWritable;
+import org.apache.hadoop.hive.serde2.io.DoubleWritable;
+import org.apache.hadoop.hive.serde2.io.ShortWritable;
+import org.apache.hadoop.hive.serde2.io.TimestampWritable;
+import org.apache.hadoop.hive.serde2.objectinspector.SettableStructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StructField;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.io.ArrayPrimitiveWritable;
+import org.apache.hadoop.io.BooleanWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.MapWritable;
+import org.apache.hadoop.io.ObjectWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.kafka.connect.data.Date;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.data.Time;
+import org.apache.kafka.connect.data.Timestamp;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public final class OrcUtil {
+
+  /**
+   * Create an object of OrcStruct given a type string and a list of objects
+   */
+  @SuppressWarnings("unchecked")
+  public static OrcStruct createOrcStruct(TypeInfo typeInfo, Object... objs) {
+    SettableStructObjectInspector oi = (SettableStructObjectInspector) 
+            OrcStruct.createObjectInspector(typeInfo);
+
+    List<StructField> fields = (List<StructField>) oi.getAllStructFieldRefs();
+    OrcStruct result = (OrcStruct) oi.create();
+    result.setNumFields(fields.size());
+    for (int i = 0; i < fields.size(); i++) {
+      oi.setStructFieldData(result, fields.get(i), objs[i]);
+    }
+
+    return result;
+  }
+
+  /**
+   * Convert a Struct into a Writable array
+   */
+  public static Object[] convertStruct(Struct struct) {
+    List<Object> data = new LinkedList<>();
+    for (Field field : struct.schema().fields()) {
+      if (struct.get(field) == null) {
+        data.add(null);
+      } else {
+        Schema.Type schemaType = field.schema().type();
+        String name = field.name();
+        switch (schemaType) {
```

Comment:
This method with this long `switch` statement is a bit old-fashioned. 
It's also responsible for 3 checkstyle suppressions. 

Here's a suggestion that will work nicely instead can will make this code more readable and will need only one suppression (can't do much about `ClassDataAbstractionCoupling` - you need what you need). 

1. Declare an `EnumMap` as `private static EnumMap<Schema.Type, BiFunction<Struct, Field, Object>> conversions` 
2. Initialize it statically. E.g. 
```
static {
   conversions.put(BOOLEAN, OrcUtil::convertBoolean);
   ...
   conversions.put(MAP, OrcUtil::convertMap);
}
```
3. Discard your switch statement altogether and instead call: 
`data.add(conversions.get(schemaType).apply(struct, field));`
  (I assumed there's no type missing, but write appropriately if that's not the case)
4. Encapsulate the code for each conversion in the respective method. E.g. 
```
private static Object convertBoolean(Struct struct, Field field) {
    return new BooleanWritable(struct.getBoolean(field.name()));
  }
```
Remove the suppressions that are not required after the refactoring.

Reply:
addressed